### PR TITLE
[woot!] Route a channel to send banks from replay to banking

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

The TPU/TVU mode transition is cumbersome and isn't necessary.  The replay stage can feed the banking stage and downstream units banks.  PohRecorder already guards against entries being recorded into those banks. 

#### Summary of Changes

Replay stage can feed the banking stage a bank directly.  This change has the following cascading effects:

1. Fullnode doesn't have a mode, so no rotation.
2. Banking stage always runs
3. Replay stage decides when to set a bank to generate entries
4. poh recorder ensures the bank only generates entries at the right time
5. Without a bank, banking stage forwards verified packets (there should probably be some QoS here or a fee check, or we are doing a blind proxy which is really easy to exploit for ddos).
5. Broadcast stage receives this ugly event, Bank + Entries.  While the data structure is ugly, it is kind of nice to have it all there so there is no synchronization step.

Fixes #

### CODE REVIEW

File: src/leader_confirmation_service.rs [hunk 1]
```diff
@@ -22,15 +21,15 @@ pub enum ConfirmationError {
 pub const COMPUTE_CONFIRMATION_MS: u64 = 100;
 
 pub struct LeaderConfirmationService {
-    thread_hdl: JoinHandle<()>,
+    current_bank: Arc<Mutex<Option<Arc<Bank>>>>, // super ugly!!!
```
Comment (garious) [inline src/leader_confirmation_service.rs]: If clippy allows this, we need to fire clippy.

Comment (aeyakovenko) [reply src/leader_confirmation_service.rs]: yea, really bad :).  i think this leader confirmation service needs to be killed completely.  but I think we can do that later

File: src/poh_service.rs [hunk 2]
```diff
@@ -108,134 +108,134 @@ impl Service for PohService {
...
-        //WorkingBank should be dropped by the PohService thread as well
-        assert_eq!(entry_receiver.recv(), Err(RecvError));
-
-        exit.store(true, Ordering::Relaxed);
-        poh_service.exit();
-        let _ = poh_service.join().unwrap();
-    }
-}
+// #[cfg(test)]
+// mod tests {
```
Comment (garious) [inline src/poh_service.rs]: what happened here?

Comment (aeyakovenko) [reply src/poh_service.rs]: oh weird.  bad edit

File: src/leader_confirmation_service.rs [hunk 3]
```diff
@@ -96,36 +91,34 @@ impl LeaderConfirmationService {
...
                 loop {
                     if exit.load(Ordering::Relaxed) {
                         break;
                     }
-                    Self::compute_confirmation(
-                        &bank,
-                        leader_id,
-                        &mut last_valid_validator_timestamp,
-                    );
+                    // dont hold this lock to long
```
Comment (garious) [inline src/leader_confirmation_service.rs]: too

File: src/leader_confirmation_service.rs (hunk 1 above)
Comment (garious) [reply src/leader_confirmation_service.rs]: Yeah, you can kill it, if in the way. It's not critical and lacks precision.

Comment (aeyakovenko) [reply src/leader_confirmation_service.rs]: @garious did some work to unhack that

**Comment by aeyakovenko** [comment]
see #3080

**Pull request closed without merging**